src/Service/GroupsService.php

```diff
@@ -33,6 +37,59 @@ public function __construct(
         parent::__construct($repository, $mapper, $cache);
     }
 
+    /**
+     * @param CoreEntity $coreEntity
+     * @param string $groupType
+     * @param int|null $limit
+     * @param int $page
+     * @param string $ttl
+     * @param string $nullTtl
+     * @return Group[]
+     */
+    public function findByCoreEntityMembership(
+        CoreEntity $coreEntity,
+        string $groupType = 'Group',
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE,
+        $ttl = CacheInterface::NORMAL,
+        $nullTtl = CacheInterface::NORMAL
+    ): array {
+        if (!($coreEntity instanceof Programme || $coreEntity instanceof Collection || $coreEntity instanceof Gallery)) {
```

Comment:
Any Group /could/ be a member of another Group, I don't think it's worth having this check here.